osquery/tables/events/yara_events.cpp

```diff
@@ -128,17 +139,37 @@ Status YARAEventSubscriber::Callback(const FileEventContextRef& ec,
   r["strings"] = std::string("");
   r["tags"] = std::string("");
 
-  ConfigDataInstance config;
-  const auto& parser = config.getParser("yara");
-  if (parser == nullptr)
+  std::shared_ptr<ConfigParserPlugin> parser;
+  try {
+    parser = Config::getParser("yara");
+  } catch (const std::bad_cast& e) {
+    return Status(1, "Error casting yara config parser plugin");
+  }
+  if (parser == nullptr || parser.get() == nullptr) {
     return Status(1, "ConfigParser unknown.");
-  const auto& yaraParser = std::static_pointer_cast<YARAConfigParserPlugin>(parser);
+  }
+
+  std::shared_ptr<YARAConfigParserPlugin> yaraParser;
+  try {
+    yaraParser = std::dynamic_pointer_cast<YARAConfigParserPlugin>(parser);
+  } catch (const std::bad_cast& e) {
+    return Status(1, "Error casting yara config parser plugin");
+  }
+  if (yaraParser == nullptr || yaraParser.get() == nullptr) {
+    return Status(1, "Yara parser unknown.");
+  }
+
   auto rules = yaraParser->rules();
 
   // Use the category as a lookup into the yara file_paths. The value will be
   // a list of signature groups to scan with.
   auto category = r.at("category");
-  const auto& yara_config = config.getParsedData("yara");
+  pt::ptree yara_config;
+  try {
+    yara_config = Config::getParser("yara")->getData();
```

Comment:
I would love for this use of the `getData` API to not require an exception check.